iam/src/configure-iam.js

```diff
@@ -45,26 +46,27 @@ const configureIAM = async (
 
   const promises = [];
   const cluster = await getClusterInfo(projectId);
+  await authenticateKubeCtl(cluster);
 
   services.forEach(({ name: namespace, repository }) => {
     const systemName = `${permissionPrefix}.${namespace}-${env}`;
 
     // 1. Authenticate Kubectl and create namespace (if not exists)
```

Comment:
I guess what you've really done is change the order things are done? So maybe update the comment too?

Reply:
Yes since we called create namespace for every service in iam yaml we connected to a cluster multiple times which led to connection timeout when creating namespace since some processes were still trying to connect to cluster. Now we only create namespace if the System doesn't exist.

If the action failed to create a system it resulted in a failure which led to envoy-opa config never being applied and services that are deployed wouldn't be able to start.

I have updated the comments! 👍